models/auth/oauth2.go

```diff
@@ -138,6 +138,9 @@ func (app *OAuth2Application) TableName() string {
 // ContainsRedirectURI checks if redirectURI is allowed for app
 func (app *OAuth2Application) ContainsRedirectURI(redirectURI string) bool {
 	contains := func(s string) bool {
+		if idx := strings.Index(s, "?"); idx > 0 {
```

Comment:
Is this right? I think we need to use `url.Parse`.

Reply:
> Is this right? I think we need to use `url.Parse`.

https://github.com/golang/go/blob/20f052c83c380b8b3700b7aca93017178a692d78/src/net/url/url.go#L535